### ISSUE TITLE
clang-format option to insert a space before () in operator overloading

### ISSUE THREAD

**Issue by cb252c30-9e88-4eb4-891d-0e28e2d53015** [llvm/llvm-project#41763]
|  |  |
| --- | --- |
| Bugzilla Link | [42418](https://llvm.org/bz42418) |
| Version | unspecified |
| OS | Windows NT |

## Extended Description 
I usually do not insert any space between function name and ().
However, I always insert a space for operator overloading only because without space, consecutive symbols look too dense to me.

That is, I prefer

```
class Test {
    auto func() -> int; // no space before ()
    auto operator++ (int) -> int; // a space before ()
};
```

to

```
class Test {
    auto func() -> int;
    auto operator++(int) -> int;
};
```

It would be nice to have an option for this formatting in clang-format.

**Comment by mydeveloperday** [comment]
This would make a good first issue, by using the `SpaceBeforeParensOptions` construct you could easily add a `BeforeOperator`

**Comment by rajdroid** [comment]
Hi, may I work on this issue?

**Comment by mydeveloperday** [comment]
Go for it.. add yourself as the assignee

**Comment by rajdroid** [comment]
@mydeveloperday I'm sorry, but I believe I don't have rights to assign an issue.
Could you please assign?

**Comment by rajdroid** [comment]
@mydeveloperday I have raised a code review (https://reviews.llvm.org/D116283) and added you as a code reviewer. Let me know if it is fine or should I add someone else as well.

**Comment by rajdroid** [comment]
@mydeveloperday can we somehow close this issue, as the fix has been merged?

**Comment by mydeveloperday** [comment]
We should be adding the "Fixes #41763" to the commit message to automatically close issues, but I can close this out.

**Comment by mydeveloperday** [comment]
Close by commit da6b0d0b768e3ecb1af2fd9df2d98510f7aff45c
